tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * WordPress dependencies
+ */
+import { activatePlugin, createURL, visitAdminPage } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	deactivateAllOtherPlugins,
+	resetSiteKit,
+	useRequestInterception,
+} from '../../../utils';
+
+describe( 'setting up the Analytics module with no existing account and no existing tag', () => {
+	beforeAll( async() => {
+		await page.setRequestInterception( true );
+		useRequestInterception( request => {
+			if ( request.url().startsWith( 'https://accounts.google.com/o/oauth2/auth' ) ) {
+				request.respond( {
+					status: 302,
+					headers: {
+						location: createURL( '/', [
+							'oauth2callback=1',
+							'code=valid-test-code',
+							'e2e-site-verification=1',
+							'scope=TEST_ALL_SCOPES',
+						].join( '&' ) )
+					}
+				} );
+			} else if ( request.url().match( '/wp-json/google-site-kit/v1/data/' ) ) {
+				request.respond( {
+					status: 200
+				} );
+			} else {
+				request.continue();
+			}
+		} );
+	} );
+
+	beforeEach( async() => {
+		await activatePlugin( 'e2e-tests-auth-plugin' );
+		await activatePlugin( 'e2e-tests-site-verification-plugin' );
+		await activatePlugin( 'e2e-tests-oauth-callback-plugin' );
+		await activatePlugin( 'e2e-tests-module-setup-analytics-api-mock-no-account' );
+	} );
+
+	afterEach( async() => {
+		await deactivateAllOtherPlugins();
+		await resetSiteKit();
+	} );
+
+	it( 'display create account when user has no analytics account', async() => {
+
+		await visitAdminPage( 'admin.php', 'page=googlesitekit-settings' );
+		await page.waitForSelector( '.mdc-tab-bar' );
+		await expect( page ).toClick( '.mdc-tab', { text: /connect more services/i } );
+		await page.waitForSelector( '.googlesitekit-settings-connect-module--analytics' );
+
+		await Promise.all( [
+			page.waitForSelector( '.googlesitekit-setup-module__action .mdc-button' ),
+			expect( page ).toClick( '.googlesitekit-cta-link', { text: /set up analytics/i } ),
+		] );
+
+		// Intercept the call to window.open and call our API to simulate a created account.
+		await page.evaluate( () => {
```

Comment:
Is there any other way to do this without overriding `window.open`; perhaps by listening to `targetcreated`?

Reply:
I looked into it, but there is a known issue with intercepting requests in new tabs in Puppeteer.

See https://github.com/GoogleChrome/puppeteer/issues/386#issuecomment-343059315

> We currently don't have a way to attach to targets as soon as they are created, so there is no way to set up request interception for new tabs that won't miss some early requests. This is a known issue.